src/all/comicgrowl/src/eu/kanade/tachiyomi/extension/all/comicgrowl/ComicGrowl.kt

```diff
@@ -1,63 +1,330 @@
+    override fun popularMangaNextPageSelector() = null
 
-    override fun popularMangaFromElement(element: Element) = SManga.create().apply {
-        title = element.select("h5").text()
-        thumbnail_url = element.select("div > img").attr("data-src")
-        setUrlWithoutDomain(element.attr("href"))
+    override fun popularMangaSelector() = ".ranking-item"
+
+    override fun popularMangaFromElement(element: Element): SManga {
+        return SManga.create().apply {
+            url = element.selectFirst("a")!!.attr("href")
+            title = element.selectFirst(".title-text")!!.text()
+            author = element.selectFirst(".author-link")!!.text()
+            thumbnail_url = getImageFromElement(element)
+        }
     }
 
-    override fun latestUpdatesSelector() =
-        "div[class=\"update latest\"] > div.card-board > " + "div[class~=card]:not([class~=ad]) > div > a"
+    override fun mangaDetailsRequest(manga: SManga) = GET(manga.url)
 
-    override fun latestUpdatesFromElement(element: Element) = SManga.create().apply {
-        title = element.select("div.data h3").text()
-        thumbnail_url = element.select("div.thumb-container img").attr("data-src")
-        setUrlWithoutDomain(element.attr("href"))
+    override fun mangaDetailsParse(document: Document): SManga {
+        val infoElement = document.selectFirst(".series-h-info")!!
+        val updateDateElement = infoElement.selectFirst("series-h-tag-label")
+        return SManga.create().apply {
+            title = infoElement.select("h1 > span")[1]!!.text()
+            author = infoElement.selectFirst(".series-h-credit-user-item > .article-text")!!.text() // TODO: get 脚本/漫画
+            description = infoElement.selectFirst(".series-h-credit-info-text-text > div > p > span > span")!!.text()
+            thumbnail_url = getImageFromElement(document.getElementsByClass("series-h-img").first()!!)
+            status = if (updateDateElement != null) SManga.ONGOING else SManga.COMPLETED
+        }
     }
 
-    override fun getCollections(): List<Collection> = listOf(
-        Collection("連載作品", ""),
-    )
+    override fun chapterListRequest(manga: SManga) = GET(manga.url) // TODO: get chapters from `/list`
 
-    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
-        if (query.isNotEmpty()) {
-            val url = "$baseUrl/search".toHttpUrl().newBuilder().addQueryParameter("q", query)
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val document = response.asJsoup()
+        return document.select(chapterListSelector()).mapIndexed { index, element ->
+            chapterFromElement(element).apply {
+                chapter_number = index.toFloat()
+                if (url.isEmpty()) { // Login required
+                    throw Exception("Some chapters don't have url, login and refresh again")
```

Comment:
User doesn't care about the exact reason. Can simplify:
```suggestion
                    throw Exception("Missing chapters, login and refresh again")
```

Reply:
Except throw exception, how can I communicate with UI layer to display snackbar or something to notify user that they need login to see missing chapters?

Can I add somewhat "dummy" url for chapters that need login or behind paywall, and add icons in the chapter title?

Reply:
Ah, in that case you would use a setting to toggle if paid chapters should be visible. If hidden, the user is not notified on missing chapters. If visible, you show the chapters with 🔒 or similar.

A dummy URL works and just error (with `throw`) if the user does not have access. I think you should use a unique URL for each, otherwise it might mark all locked chapters as read if one of them is read.

Reply:
OK, I override `pageListRequest` to check if it's a dummy url and throw exception.